src/components/VersionSwitch.tsx

```diff
@@ -0,0 +1,45 @@
+'use client'
+
+import * as DropdownMenu from '@radix-ui/react-dropdown-menu'
+import { ChevronDownIcon } from 'lucide-react'
+import { NavLinkButton } from './ui/NavLinkButton'
+import { NavLink } from './NavLink'
+import { Tag } from './ui/Tag'
+import { VERSIONS, CURRENT_VERSION } from '@/utils/constants'
+import type { VersionData } from '@/types'
+
+const VersionItem = ({ version }: { version: VersionData }) => {
+  return (
+    <DropdownMenu.Item asChild>
+      <NavLink href={version.url} target={version.version !== CURRENT_VERSION ? '_blank' : ''}>
+        {version.version}
+        {version.isLegacy ? <Tag className="ml-1">Legacy</Tag> : null}
+        {version.isBeta ? <Tag className="ml-1">Beta</Tag> : null}
+        {version.isAlpha ? <Tag className="ml-1">Alpha</Tag> : null}
+        {version.isRc ? <Tag className="ml-1">RC</Tag> : null}
```

Comment:
Refactor: Tag component is repeated 4 times, you can extract a reusable function